llama_index/chat_engine/types.py

```diff
@@ -28,7 +29,7 @@ def __str__(self) -> str:
                 self.response_str += delta
         return self.response_str
 
-    def write_response_to_history(self, chat_history: List[ChatMessage]) -> None:
+    def write_response_to_history(self, memory: BaseMemory) -> None:
```

Comment:
@logan-markewich Hi, Logan:
May I wonder what is the design consideration of this function?
Does `StreamingChatResponse` takes too much work? It now requires an extra thread to call this function, to get `response_gen` working.
In response_gen, the while loop would consume CPU time until queue is been empty.
Should we just use simple `StreamingResponse`,  and move memory management to agent?

Reply:
Can we yield `chat_stream` in `response_gen' and save memory in the end?
Or still, handle memory in agent by wrapping response_gen before user exhaust it. that what you taught me.
The thing is current implement make it very hard to wrap `response_gen` within `StreamingChatResponse`. 
It's a @property method which do not allow modification

Reply:
@logan-markewich 
For instance, will the following code work?:
```python
@property
def response_gen(self) -> Generator[str, None, None]:
    for chat in chat_stream:
        delta = chat.delta
        self._unformatted_response += delta
        yield delta
    self.memory += memory.put(chat.message)
    if self.callback != None:
        self.callback(chat) # this is what I want to implement, call_back also need full message
```